src/odb_backend.c

```diff
@@ -0,0 +1,348 @@
+/*
+ * Copyright 2010-2019 The pygit2 contributors
+ *
+ * This file is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License, version 2,
+ * as published by the Free Software Foundation.
+ *
+ * In addition to the permissions in the GNU General Public License,
+ * the authors give you unlimited permission to link the compiled
+ * version of this file into combinations with other programs,
+ * and to distribute those combinations without any restriction
+ * coming from the use of this file.  (The General Public License
+ * restrictions do apply in other respects; for example, they cover
+ * modification of the file, and distribution when not linked into
+ * a combined executable.)
+ *
+ * This file is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; see the file COPYING.  If not, write to
+ * the Free Software Foundation, 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301, USA.
+ */
+
+#define PY_SSIZE_T_CLEAN
+#include <Python.h>
+#include "error.h"
+#include "object.h"
+#include "oid.h"
+#include "types.h"
+#include "utils.h"
+#include <git2/odb_backend.h>
+#include <git2/sys/alloc.h>
+#include <git2/sys/odb_backend.h>
+
+static int
+OdbBackend_build_as_iter(const git_oid *oid, void *accum)
+{
+    int err;
+    PyObject *py_oid = git_oid_to_python(oid);
+
+    err = PyList_Append((PyObject*)accum, py_oid);
+    Py_DECREF(py_oid);
+    return err;
```

Comment:
Return `GIT_EUSER` if there's an error.
`PyList_Append` will return -1 on error, which would be interpreted as `GIT_ERROR` in `OdbBackend_as_iter`, and the exception set by `PyList_Append` would be overwritten.

More information in https://github.com/libgit2/libgit2/issues/4783

`Odb_build_as_iter` has the same problem, please fix it there as well.